app/src/main/java/com/wittyapp/MainActivity.kt

```diff
@@ -0,0 +1,54 @@
+package com.wittyapp
+
+import android.content.Intent
+import android.os.Bundle
+import android.support.v7.app.AppCompatActivity
+import android.util.Log
+import android.view.View
+import android.widget.Button
+import com.facebook.CallbackManager
+import com.facebook.FacebookCallback
+import com.facebook.FacebookException
+import com.facebook.login.LoginManager
+import com.facebook.login.LoginResult
+import java.util.*
+
+class MainActivity : AppCompatActivity() {
+    private var callbackManager: CallbackManager? = null
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_main)
+
+        val btnLoginFacebook = findViewById<Button>(R.id.btnLoginFacebook)
+
+        btnLoginFacebook.setOnClickListener({
+            // Login
+            callbackManager = CallbackManager.Factory.create()
+            LoginManager.getInstance().logInWithReadPermissions(this, Arrays.asList("public_profile", "email"))
+            LoginManager.getInstance().registerCallback(callbackManager,
+                    object : FacebookCallback<LoginResult> {
+                        override fun onSuccess(loginResult: LoginResult) {
+                            Log.d("MainActivity", "Facebook token: " + loginResult.accessToken.token)
```

Comment:
Let's use SLF4J. This will allow us to easily swap logging destinations (eg. local text file or logging to the server via REST APIs, instead of the default Android log). This should be done as early as possible because it becomes a pain later on to change every single log statement to match the correct syntax.

Read this: https://www.baeldung.com/kotlin-logging
then realize its too much trouble and just use this: https://github.com/MicroUtils/kotlin-logging